### PR TITLE
openvpn: update to 2.6.5

### PR DESCRIPTION
**Pull request by blackzafiqz**
Maintainer: @mkrkn
Compile tested: mt7622, snapshot
Run tested: Xiaomi AX3200

Updated to 2.6.5

### CODE REVIEW

**Comment by blackzafiqz** [comment]
@mkrkn  can you approve test build again?

**Comment by AuthorReflex** [comment]
compile && runtime tested builds:
ramips mt7621/Xiaomi Mi router pro - openssl variant
ramips mt7620/TP-Link Archer C50 - mbedtls variant
ramips mt7621/Beeline Smartbox Giga - wolfssl variant

MiR3P is a server, many clients (windows, linux, embedded) works without problems
also, MiR3P is a client to vps in another country
Everything works fine.

**Comment by AuthorReflex** [comment]
And 2.6.5 is out
no changes in patches requred

**Comment by blackzafiqz** [comment]
I will update it tonight.

**Comment by blackzafiqz** [comment]
@AuthorReflex may i know why github build throw up: Dirty patches detected, please refresh and review the diff. I already tried make package/openvpn/{clean,refresh}

**Comment by AuthorReflex** [comment]
> @AuthorReflex may i know why github build throw up: Dirty patches detected, please refresh and review the diff. I already tried `make package/openvpn/{clean,refresh}`

cause is described in my review (in patch file removed lines must be ```---a``` added lines must be ```+++b```) in patch 101 it's wrong

**Comment by blackzafiqz** [comment]
Thank you, fixed and pushed.

**Comment by AuthorReflex** [comment]
> Thank you, fixed and pushed.

still errors here
[101-wolfssl-disable-usage-unimplemented-functions.patch]
```
--- a/src/openvpn/crypto_openssl.c
+++ a/src/openvpn/crypto_openssl.c
```
+++ must be "b"

**Comment by blackzafiqz** [comment]
my bad, didn't staged that file. thank you sir.

**Comment by AuthorReflex** [comment]
Refreshed && tested on my devices. LGTM

**Comment by AuthorReflex** [comment]
@blackzafiqz  you need to push file 101-wolfssl-disable-usage-unimplemented-functions.patch after refreshing

File: net/openvpn/Config-mbedtls.in [hunk 1]
```diff
@@ -24,26 +24,23 @@ config OPENVPN_mbedtls_ENABLE_FRAGMENT
 	bool "Enable internal fragmentation support (--fragment)"
 	default y
 
-config OPENVPN_mbedtls_ENABLE_MULTIHOME
```
Comment (BKPepe) [inline net/openvpn/Config-mbedtls.in:27]: Why this was removed? I can not see anything in the commit description.

File: net/openvpn/Makefile [hunk 2]
```diff
@@ -36,14 +36,14 @@ define Package/openvpn/Default
   URL:=http://openvpn.net
   SUBMENU:=VPN
   MENU:=1
-  DEPENDS:=+kmod-tun +OPENVPN_$(1)_ENABLE_LZO:liblzo +OPENVPN_$(1)_ENABLE_IPROUTE2:ip $(3)
+  DEPENDS:=+kmod-tun +libcap-ng +OPENVPN_$(1)_ENABLE_LZO:liblzo +OPENVPN_$(1)_ENABLE_LZ4:liblz4 +OPENVPN_$(1)_ENABLE_IPROUTE2:ip +OPENVPN_$(1)_ENABLE_DCO:libnl-genl $(3)
```
Comment (BKPepe) [inline net/openvpn/Makefile:39]: dtto

File: net/openvpn/Makefile [hunk 3]
```diff
@@ -9,14 +9,14 @@ include $(TOPDIR)/rules.mk
...
-PKG_RELEASE:=3
+PKG_VERSION:=2.6.5
+PKG_RELEASE:=1
 
 PKG_SOURCE_URL:=\
 	https://build.openvpn.net/downloads/releases/ \
 	https://swupdate.openvpn.net/community/releases/
-PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
-PKG_HASH:=2bbd0026469902037ee6499b68283d5ab36c74e36cae3112082cfdf6c77a0c57
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
```
Comment (BKPepe) [inline net/openvpn/Makefile:18]: dtto. It appears that upstream changed the compression. There should be something in the commit description about that.

**Review by BKPepe — CHANGES_REQUESTED** [review]

File: net/openvpn/patches/101-wolfssl-disable-usage-unimplemented-functions.patch [hunk 4]
```diff
@@ -0,0 +1,31 @@
+--- a/src/openvpn/crypto_openssl.c
++++ b/src/openvpn/crypto_openssl.c
+@@ -51,7 +51,7 @@
```
Comment (AuthorReflex) [inline net/openvpn/patches/101-wolfssl-disable-usage-unimplemented-functions.patch:3]:
```
@@ -1419,7 +1419,7 @@ engine_load_key(const char *file, SSL_CT
 #endif /* if HAVE_OPENSSL_ENGINE */
 }
 ```
 incorrect patch place causes an error

**Comment by jefferyto** [comment]
I think your commit message needs to explain the changes you are making.

**Comment by blackzafiqz** [comment]
i will update everything tomorrow

**Comment by sch-m** [comment]
@blackzafiqz Are you still working on this PR?

I am also interested in getting openvpn-2.6.x with DCO support in and did some work here:
https://github.com/openwrt/packages/compare/master...TDT-AG:packages:pr/20230628-openvpn-dco

**Comment by danny30au** [comment]
> @blackzafiqz Are you still working on this PR?
> 
> I am also interested in getting openvpn-2.6.x with DCO support in and did some work here: [master...TDT-AG:packages:pr/20230628-openvpn-dco](https://github.com/openwrt/packages/compare/master...TDT-AG:packages:pr/20230628-openvpn-dco)

Hopefully we can get this "Pushed" to master soonish the (offload kmod) will be very beneficial to the performance of OpenVPN tunnels and cpu usage for users :)

**Pull request closed without merging**